server/src/main/java/io/crate/execution/ddl/tables/AddColumnRequest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.ddl.tables;
+
+import io.crate.metadata.Reference;
+import io.crate.metadata.RelationName;
+import io.crate.sql.tree.CheckColumnConstraint;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Contains all supported for ADD COLUMN configurations such as constraints,
+ * doc values enabled flag, column type properties (bit length, text analyzer e.t.c).
+ */
+public class AddColumnRequest extends AcknowledgedRequest<AddColumnRequest> {
+
+    private final RelationName relationName;
+    private final boolean isPartitioned;
+    private final Reference columnRef; // Contains isNullable and hasDocValues which reflects NOT-NULL and STORAGE WITH.
+    private final Map<String, Object> columnProperties = new LinkedHashMap<>();
```

Comment:
question: why is this `LinkedHashMap`?

If this is by any chance because of my commits regarding column ordering, it is unnecessary for `add column` since the negative 'column estimates' take care of it.

Reply:
I'm currently addressing a comment to get rid of this map and use bunch of concrete fields instead, so it will be gone.
But originally I used to merge maps as-is and aligned with data structures I saw in runtime (I think it's defined in DocIndexMetadata). Related - https://github.com/crate/crate/pull/12806#discussion_r940166798